src/v/archival/ntp_archiver_service.cc

```diff
@@ -341,7 +361,7 @@ ss::future<> ntp_archiver::upload_until_abort() {
         bool is_synced = co_await _parent.archival_meta_stm()->sync(
           sync_timeout);
         if (!is_synced) {
-            co_return;
+            continue;
```

Comment:
`sync` on its own doesn't do anything though. It just creates a waiter. The application of batches is done by the `state_machine_manager`.

Reply:
Oh perhaps you meant going through the loop body every 10 seconds. I think it would be fine as the loop doesn't do anything and 10s is a pretty long time.

Reply:
What do you mean by 10s? The upload loop uses exponential backoff with 10s limit to wait if there is nothing to upload but it's waiting after the sync. And in case of sync, do we have any guarantees that in case of failure it will always wait for timeout? 

The failure I'm thinking about is:
- sync returns error immediately without waiting for timeout
- then it gets retried immediately 
- rinse, repeat

Reply:
`upload_loop_until_term_change` uses exponential back-off, but `upload_until_abort` does not. If the nodes is the leader, then sync should wait for the application. I don't think the scenario you're describing is possible because the loop waits for leadership at the top via a cvar.